packages/application-shell/src/components/navbar/navbar.tsx

```diff
@@ -0,0 +1,667 @@
+    hasMenuEveryMenuVisibilitySetToBeHidden(
+      props.menuVisibilities,
+      props.namesOfMenuVisibilities
+    ) ||
+    isMenuItemDisabledForEnvironment(
+      props.keyOfMenuItem,
+      props.disabledMenuItems
+    )
+  )
+    return null;
+
+  const permissionsWrapper =
+    (Array.isArray(props.permissions) && props.permissions.length > 0) ||
+    (Array.isArray(props.dataFences) && props.dataFences.length > 0) ? (
+      <RestrictedByPermissions
+        permissions={props.permissions}
+        actionRights={props.actionRights}
+        dataFences={props.dataFences}
+        selectDataFenceData={demandedDataFence => {
+          switch (demandedDataFence.type) {
+            case 'store':
+              return demandedDataFence.actualDataFenceValues;
+            default:
+              return null;
+          }
+        }}
+        // Always check that some of the given permissions match.
+        shouldMatchSomePermissions={true}
+      >
+        {props.children}
+      </RestrictedByPermissions>
+    ) : (
+      <>{props.children}</>
+    );
+
+  if (props.featureToggle) {
+    return (
+      <ToggleFeature flag={props.featureToggle}>
+        {permissionsWrapper}
+      </ToggleFeature>
+    );
+  }
+  return permissionsWrapper;
+};
+RestrictedMenuItem.displayName = 'RestrictedMenuItem';
+RestrictedMenuItem.defaultProps = restrictedMenuItemDefaultProps;
+
+export const getIconColor = (
+  isActive: boolean,
+  isAlternativeTheme: boolean
+) => {
+  if (isActive) return 'primary40';
+  return isAlternativeTheme ? 'neutral60' : 'surface';
+};
+
+const getMenuVisibilitiesOfSubmenus = (menu: TNavbarMenu) =>
+  menu.submenu.map(submenu => submenu.menuVisibility).filter(nonNullable);
+const getMenuVisibilityOfMainmenu = (menu: TNavbarMenu) =>
+  menu.menuVisibility ? [menu.menuVisibility] : [];
+
+type ApplicationMenuProps = {
+  location: RouteComponentProps['location'];
+  menuType: string;
+  menu: TNavbarMenu;
+  isActive: boolean;
+  isMenuOpen: boolean;
+  shouldCloseMenuFly: MouseEventHandler<HTMLElement>;
+  menuVisibilities?: TApplicationContext<{}>['menuVisibilities'];
+  handleToggleItem: (menuType: string, index: string) => void;
+  applicationLocale: string;
+  projectKey: string;
+  disabledMenuItems?: string[];
+  useFullRedirectsForLinks?: boolean;
+  onMenuItemClick?: MenuItemLinkProps['onClick'];
+};
+const ApplicationMenu = (props: ApplicationMenuProps) => {
+  const hasSubmenu =
+    Array.isArray(props.menu.submenu) && props.menu.submenu.length > 0;
+  const namesOfMenuVisibilitiesOfAllSubmenus = hasSubmenu
+    ? getMenuVisibilitiesOfSubmenus(props.menu)
+    : getMenuVisibilityOfMainmenu(props.menu);
+
+  const isMainMenuRouteActive = (link: string) => {
+    const match = matchPath(props.location.pathname, {
+      path: `/${props.projectKey}/${link}`,
+      exact: false,
+      strict: false,
+    });
+    return Boolean(match);
+  };
+  const renderLabel = (labelAllLocales: TLocalizedField[]) => {
```

Comment:
This was like that right? Feels more like a component than a render prop.

Reply:
Yeah, you're right: https://github.com/commercetools/merchant-center-application-kit/pull/1273/commits/ccf8cc5bea3ba9edff296057a6e49d7fe19e0be5